mathesar/serializers.py

```diff
@@ -93,16 +95,20 @@ class DataFileSerializer(serializers.ModelSerializer):
         default=serializers.CurrentUserDefault(), read_only=True
     )
     header = serializers.BooleanField(default=True)
+    paste = serializers.CharField(required=False, trim_whitespace=False)
```

Comment:
I don't see a field on the model that stores how the file was created. You should store something on the model so that you can always tell when using a data file object whether it was created by uploading a file or pasting it. For example, we might one day want to have a page that shows all the files that the user has created, including information on whether it was uploaded or pasted in.

 I'd also recommend naming the field on the model something a little more verbose like `from_paste` or `created_from_paste` if it's a boolean. Alternatively, you could also make it a `creation_method` field with choices of paste or upload (that's extendable to other methods, like getting it from a remote URL).